g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/dao/impl/S3OneComponentDaoImpl.java

```diff
@@ -0,0 +1,150 @@
+      } catch (IOException e) {
+         String errorLog = ConstantsKeys.FATA_ERROR + e.getMessage();
+         logger.error(errorLog, e);
+         throw new DataException("File is not existing: ");
+      }
+      if (result != null) {
+         result.setProduct(productName);
+         result.setVersion(version);
+         result.setComponent(component);
+         result.setLocale(locale);
+      } else {
+         throw new DataException("File is not existing: ");
+      }
+      return result;
+   }
+
+   /**
+    * get one compose bundle files from s3 server as json String
+    */
+   @Override
+   public String get2JsonStr(String productName, String version, String component, String locale)
+         throws DataException {
+      String filePath = S3Utils.genProductVersionS3Path(productName, version) + component
+            + S3Utils.S3FILE_SEPARATOR + ResourceFilePathGetter.getLocalizedJSONFileName(locale);
+      String result = null;
+      if (s3Client.getS3Client().doesObjectExist(config.getBucketName(), filePath)) {
+         S3Object o = s3Client.getS3Client().getObject(config.getBucketName(), filePath);
+         if (o != null) {
+            try {
+               result = S3Utils.S3Obj2Str(o);
+            } catch (IOException e) {
+               logger.warn(e.getMessage(), e);
+               throw new DataException("S3File is not existing: " + filePath);
+            }
+         } else {
+            throw new DataException("S3 File is not existing: " + filePath);
+         }
+      }
+      if (result == null) {
+         throw new DataException("S3 File is not existing: " + filePath);
+      }
+      return result;
+   }
+
+   @Override
+   public boolean add(String productName, String version, String component, String locale,
+         Map<String, String> messages) throws DataException {
+      return false;
+   }
+
+   /**
+    * update the compose bundle file to remote S3 server
+    */
+   @Override
+   public boolean update(String productName, String version, String component, String locale,
+         Map<String, String> messages) throws DataException {
+      if (StringUtils.isEmpty(component)) {
+         component = ConstantsFile.DEFAULT_COMPONENT;
+      }
+      String filePath = S3Utils.genProductVersionS3Path(productName, version) + component
+            + S3Utils.S3FILE_SEPARATOR + ResourceFilePathGetter.getLocalizedJSONFileName(locale);
+      Map<String, Object> json = new HashMap<String, Object>();
+      json.put(ConstantsKeys.COMPONENT, component);
+      json.put(ConstantsKeys.lOCALE, locale);
+      json.put(ConstantsKeys.MESSAGES, messages);
+      String content;
+      try {
+         content = new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(json);
+      } catch (JsonProcessingException e) {
+         throw new DataException(
+               ConstantsKeys.FATA_ERROR + "Failed to write content to file: " + filePath + ".", e);
```

Comment:
Is this really writting? I think it's converting. If so, please changet error message.